### DISCUSSION TITLE
How to view/read the used shmids of iceoryx?

### DISCUSSION THREAD

**Discussion by hannes09** [eclipse-iceoryx/iceoryx#948] [Q&A]
I want to inspect the shared memory managed by  iceoryx.
Command line tools like 
[ipcs ](https://man7.org/linux/man-pages/man1/ipcs.1.html)

dont show any shared memory segments in my setup. Why?

Output of `ipcs -m` :

`------ Shared Memory Segments --------
key        shmid      owner      perms      bytes      nattch     status`

/dev/shm/iceoryx_mgmt exist and I also have  a working ROS2 zero copy topic.

**Comment by elfenpiff** [comment]
@hannes09 

According to the man page of `ipcs` it `shows information on System V inter-process communication facilities` but iceoryx is using posix semaphores and posix shared memory.
You can find the file handles of those constructs in `/dev/shm` but I am not aware of a tool like `ipcs` for posix ipc constructs.

**Comment by hannes09** [comment]
Also `lsipc --shmem` shows no entry.

Does the  posix shared memory have segemts id? 

I want to get the shmids to work with [shmat](https://linux.die.net/man/2/shmat) on some of the shared memory.
